test_driver/app_test.dart

```diff
@@ -0,0 +1,104 @@
+      SerializableFinder text = find.text('Email');
+      await driver.waitFor(text);
+      expect(await driver.getText(text), 'Email');
+    });
+
+    test('verify the password hint text on login screen', () async {
+      SerializableFinder text = find.text('Password');
+      await driver.waitFor(text);
+      expect(await driver.getText(text), 'Password');
+    });
+
+    test('verify the text on login button', () async {
+      final buttonFinder = find.text('Log in');
+      expect(await driver.getText(buttonFinder), 'Log in');
+    });
+
+    test(
+      'verify login error message',
+      () async {
+        String emailValue = 'abcd@gmail.com';
+        SerializableFinder emailTextField =
+            find.byValueKey('emailTextFieldKey');
+        await driver.waitFor(emailTextField);
+        await driver.tap(emailTextField);
+        await driver.enterText(emailValue);
+        SerializableFinder emailText = find.text(emailValue);
+        await driver.waitFor(emailText);
+
+        String passwordValue = '12345678910';
+        SerializableFinder passwordTextField =
+            find.byValueKey('passwordTextFieldKey');
+        await driver.waitFor(passwordTextField);
+        await driver.tap(passwordTextField);
+        await driver.enterText(passwordValue);
+        SerializableFinder passwordText = find.text(passwordValue);
+        await driver.waitFor(passwordText);
+
+        SerializableFinder loginButton = find.byValueKey('loginButtonKey');
+        await driver.waitFor(loginButton);
+        await driver.tap(loginButton);
+
+        String errorMessage = 'Bad request';
+        SerializableFinder loginErrorText = find.text(errorMessage);
+        await driver.waitFor(loginErrorText);
+        expect(await driver.getText(loginErrorText), errorMessage);
+      },
+      timeout: const Timeout(
+        Duration(minutes: 2),
+      ),
+    );
+
+    test(
+      'verify login success and navigate to home screen',
+      () async {
+        String emailValue = 'kaung@nimblehq.co';
+        SerializableFinder emailTextField =
+            find.byValueKey('emailTextFieldKey');
+        await driver.waitFor(emailTextField);
+        await driver.tap(emailTextField);
+        await driver.enterText(emailValue);
+        SerializableFinder emailText = find.text(emailValue);
+        await driver.waitFor(emailText);
+
+        String passwordValue = '12345678';
+        SerializableFinder passwordTextField =
+            find.byValueKey('passwordTextFieldKey');
+        await driver.waitFor(passwordTextField);
+        await driver.tap(passwordTextField);
+        await driver.enterText(passwordValue);
+        SerializableFinder passwordText = find.text(passwordValue);
+        await driver.waitFor(passwordText);
+
+        SerializableFinder loginButton = find.byValueKey('loginButtonKey');
+        await driver.waitFor(loginButton);
+        await driver.tap(loginButton);
+
+        String homeScreenTextValue = 'Today';
```

Comment:
Can we reference them from `app_en.arb` ? 🙏

Reply:
I am confused how to use it because we don't have context like AppLocalizations.of(context)!.today